driver/marginrankingloss_driver.hpp

```diff
@@ -0,0 +1,554 @@
+    std::vector<Tref> in1Grad_host;
+    std::vector<Tref> in2Grad_host;
+
+    std::vector<int> dims;
+    float margin;
+    float divisor;
+    bool is_forward;
+    miopenMarginRakningLossReductionMode_t reduction_mode;
+};
+
+template <typename Tgpu, typename Tref>
+int MarginRankingLossDriver<Tgpu, Tref>::ParseCmdLineArgs(int argc, char* argv[])
+{
+    inflags.Parse(argc, argv);
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        miopenEnableProfiling(GetHandle(), true);
+    }
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+std::vector<int> MarginRankingLossDriver<Tgpu, Tref>::GetTensorDimsFromCmd()
+{
+    std::string lengthsStr = inflags.GetValueStr("dims");
+
+    std::vector<int> lengths;
+    std::size_t pos = 0;
+    std::size_t new_pos;
+
+    new_pos = lengthsStr.find(',', pos);
+    while(new_pos != std::string::npos)
+    {
+        std::string sliceStr = lengthsStr.substr(pos, new_pos - pos);
+
+        int len = std::stoi(sliceStr);
+
+        lengths.push_back(len);
+
+        pos     = new_pos + 1;
+        new_pos = lengthsStr.find(',', pos);
+    };
+
+    std::string sliceStr = lengthsStr.substr(pos);
+    int len              = std::stoi(sliceStr);
+
+    lengths.push_back(len);
+
+    while (lengths.size() < 5)
+    {
+        lengths.push_back(1);
+    }
+
+    return (lengths);
+}
+
+template <typename Tgpu, typename Tref>
+int MarginRankingLossDriver<Tgpu, Tref>::GetandSetData()
+{
+    dims = GetTensorDimsFromCmd();
+    SetTensorNd(input1Desc, dims, data_type);
+    SetTensorNd(input2Desc, dims, data_type);
+    SetTensorNd(targetDesc, dims, data_type);
+
+    auto reduction_mode_string = inflags.GetValueStr("reduction");
+    if(reduction_mode_string == "none")
+    {
+        reduction_mode = MIOPEN_MARGINRANKINGLOSS_REDUCTION_NONE;
+        divisor        = 0.0f;
+    }
+    else if(reduction_mode_string == "sum")
+    {
+        reduction_mode = MIOPEN_MARGINRANKINGLOSS_REDUCTION_SUM;
+        divisor        = 1.0f;
+    }
+    else if(reduction_mode_string == "mean")
+    {
+        reduction_mode = MIOPEN_MARGINRANKINGLOSS_REDUCTION_MEAN;
+        divisor        = static_cast<float>(miopen::deref(input1Desc).GetElementSize());
+    }
+    else
+    {
+        return miopenStatusInvalidValue;
+    }
+
+    margin     = inflags.GetValueDouble("margin");
+    is_forward = static_cast<bool>(inflags.GetValueInt("forw"));
+
+    if(is_forward)
+    {
+        SetTensorNd(outputDesc, dims, data_type);
+    }
+    else
+    {
+        SetTensorNd(outGradDesc, dims, data_type);
+        SetTensorNd(in1GradDesc, dims, data_type);
+        SetTensorNd(in2GradDesc, dims, data_type);
+    }
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int MarginRankingLossDriver<Tgpu, Tref>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw", 'F', "1", "MarginRankingLoss direction (Default=1)", "int");
+    inflags.AddInputFlag(
```

Comment:
First I thought you should separate those dims because they usually do in this codebase.
But I found one case similar to your try and it was @et16kr actually. He passed the PR so I guess it's fine. Check `driver/adam_driver.hpp` and try to ensure consistency with that code.

Reply:
Thanks, I've updated my code based on this PR https://github.com/ROCm/MIOpen/pull/2868/files?diff=unified&w=0#diff-ccb9bdac160d03d52ff344b10e8aced79dc056cdb4f755f6826283d07cdc4336R287